### ISSUE TITLE
Invalid syntax  result from Integrate command.

### ISSUE THREAD

**Issue by nasser1** [Mathics3/mathics-core#332]
Using 4.0 on Linux, I get the following

>mathics

```
Mathics 4.0.0
on CPython 3.10.4 (main, Apr  2 2022, 09:04:19) [GCC 11.2.0]
using SymPy 1.10.1, mpmath 1.2.1, numpy 1.22.4

Copyright (C) 2011-2021 The Mathics Team.
This program comes with ABSOLUTELY NO WARRANTY.
This is free software, and you are welcome to redistribute it
under certain conditions.
See the documentation for the full license.

Quit by evaluating Quit[] or by pressing CONTROL-D.

In[1]:= Integrate[(1 - x^3)^(2/3)/x,x]
Out[1]= --1 ^ (2 / 3) x ^ 2 Gamma[-2 / 3] hyper[{-2 / 3, -2 / 3}, {1 / 3}, 1 / x ^ 3] / (3 Gamma[1 / 3])

```

Notice the `--1` output. This is an error. Mathematica says

`PreDecrement::rvalue: 1 is not a variable with a value, so its value cannot be changed.`

Btw, as a side issue, what is `hyper[` function above? There is no such function in Mathematica.

But the main problem is generating `--1`

**Comment by rocky** [comment]
> Btw, as a side issue, what is hyper[ function above? 

It probably is SymPy's [hyper()](https://docs.sympy.org/latest/modules/functions/special.html?highlight=hyper#sympy.functions.special.hyper.hyper) and probably appears because there is no Mathematica equivalent. So this is a translation from SymPy results into the Wolfram Language. In that regard to your other bug https://github.com/mathics/Mathics/issues/1591 which is also related to conversion from SymPy into the Wolfram Language.

Personally, I don't think I will be looking at this for a while. However if you are interested in digging deeper you might check to see what would happen if you wrote this in SymPy and then compare the results.

**Comment by DUOLabs333** [comment]
This is because Mathics removes the parentheses from the result. Here is what sympy returns: `-(-1)**(2/3)*x**2*gamma(-2/3)*hyper((-2/3, -2/3), (1/3,), x**(-3))/(3*gamma(1/3))`.

**Comment by nasser1** [comment]
> removes the parentheses from the result

Thanks for finding this.  Removing parentheses from result is not good thing to do. Hopefully someone familiar with Mathics code could fix this.

**Comment by DUOLabs333** [comment]
You can get the same thing with `-Power[-1,2/3]`.

**Comment by rocky** [comment]
> Thanks for finding this. Removing parentheses from result is not good thing to do. Hopefully someone familiar with Mathics code could fix this.

No one is born being familiar with Mathics code. Mathics is an open source project. That means anyone who is interested enough in solving a problem, can do so. 

There are a number of long-standing problems in Mathics that personally, I want to address before this kind of thing.  Basically where one would start to look is in how sympy expressions are converted to Mathics expressions. `git grep -n from_sympy` will list the the places where this is done. And that other issue where there was a traceback, that function should also be replaced with a call to from_sympy() as well.

I don't doubt that these bugs will get fixed. But in the scheme of things, for me it is going to be a while before I look at such things.

**Comment by DUOLabs333** [comment]
> Basically where one would start to look is in how sympy expressions are converted to Mathics expressions. `git grep -n from_sympy` will list the the places where this is done. And that other issue where there was a traceback, that function should also be replaced with a call to from_sympy() as well.

Thanks --- I was looking through the codebase to solve this. Why would that be the culprit --- I was thinking that it had something to do with the implementation of `Power`, as it only happens to negative integers.

**Comment by rocky** [comment]
> 
> Thanks --- I was looking through the codebase to solve this. Why would that be the culprit --- I was thinking that it had something to do with the implementation of `Power`, as the bug doesn't occur with negative variables.

Note the corrected grep: `git grep -n "def from_sympy("`  which shows where the function is defined. The other grep works as well, but it shows all uses which here is too much.

**Comment by DUOLabs333** [comment]
My question still remains --- why would `from_sympy` have to do with anything.

**Comment by DUOLabs333** [comment]
Where does Mathics determine how to format `Power`? Instead of trying to make negative numbers equal to Times[-1,abs(expr)], it'll probably make more sense to just apply the parentheses rule to negative numbers as well.

**Comment by mmatera** [comment]
Something that we could start to do is to create a module to handle hypergeometic functions. You can mimic the code from the exptrig module.

**Comment by mmatera** [comment]
Regarding the format of "power" it is defined in the mathics.builtin.arithfns.basic module

**Comment by DUOLabs333** [comment]
What decides if something is wrapped in parentheses?

**Comment by DUOLabs333** [comment]
I go something that works, I just need to deal with the fact that integers don't have precision.

**Comment by DUOLabs333** [comment]
Here's my preliminary diff:
```
diff --git a/mathics/builtin/inout.py b/mathics/builtin/inout.py
index 2d241338..0f740ac4 100644
--- a/mathics/builtin/inout.py
+++ b/mathics/builtin/inout.py
@@ -212,8 +212,15 @@ def parenthesize(precedence, leaf, leaf_boxes, when_equal):
         leaf_prec = leaf.leaves[1].get_int_value()
     else:
         leaf_prec = builtins_precedence.get(leaf.get_head_name())
+    
+    #Give integer precision of Infinity
+    if isinstance(leaf,Integer):
+        leaf_prec=float("inf")
+    
+    #Check if leaf is an integer
+    is_negative = True if isinstance(leaf,Integer) and int(leaf.to_python())<0 else False
     if precedence is not None and leaf_prec is not None:
-        if precedence > leaf_prec or (precedence == leaf_prec and when_equal):
+        if precedence > leaf_prec or (precedence == leaf_prec and when_equal) or is_negative:
             return Expression(
                 SymbolRowBox,
                 Expression(SymbolList, String("("), leaf_boxes, String(")")),
```
Anything I should change before submitting a PR?

**Comment by rocky** [comment]
Does this pass all of the existing tests and doctests? And a test should be added that catches this particular problem.

A quick look and this doesn't feel right. builtins_precendence .get() should be returning the right value so there shouldn't be this test or possibly there is some larger problem.

**Comment by DUOLabs333** [comment]
In `builtins_precedence`, ``System`Integer`` isn't defined.

**Comment by rocky** [comment]
What I see is a bit different than just a missing parenthesis. The -1 should be used in a Rational[-1, 3]: 

```
$ mathics 

Mathics 4.0.1.dev0
on CPython 3.8.12 (default, Oct 21 2021, 14:26:19) 
using SymPy 1.8, mpmath 1.2.1, numpy 1.21.3, cython 0.29.28

Copyright (C) 2011-2021 The Mathics Team.
This program comes with ABSOLUTELY NO WARRANTY.
This is free software, and you are welcome to redistribute it
under certain conditions.
See the documentation for the full license.

Quit by evaluating Quit[] or by pressing CONTROL-D.

In[1]:= Integrate[(1 - x^3)^(2/3)/x,x] // FullForm
Out[1]= Times[Rational[-1, 3], Power[-1, Rational[2, 3]], Power[x, 2], Gamma[Rational[-2, 3]], Power[Gamma[Rational[1, 3]], -1], hyper[List[Rational[-2, 3], Rational[-2, 3]], List[Rational[1, 3]], Power[x, -3]]]
```

**Comment by DUOLabs333** [comment]
What do you mean?

**Comment by rocky** [comment]
I mean you may have made a mistake in you sympy translation. Paste the exact sympy session like I did above. 

I thought we went over the basics of reporting all input given and output shown before. 

I am sorry, but I do not have the time right now to be able to answer all of your questions.

**Comment by mmatera** [comment]
Here there is a simpler example:
```-1/3*(-1)^(3/4)```

The FullForm of this expression is 

```Times[Rational[-1, 3], Power[-1, Rational[3, 4]]]```

but in StandardForm, we get
```--1 ^ (3 / 4) / 3```

@rocky, I am going to try fixing this before the release.

**Comment by mmatera** [comment]
Fixed in #460
